domain/infosync/info.go

```diff
@@ -451,6 +460,91 @@ func (is *InfoSyncer) updateTopologyAliveness(ctx context.Context) error {
 		clientv3.WithLease(is.topologySession.Lease()))
 }
 
+// GetPrometheusAddr gets prometheus Address
+func GetPrometheusAddr() (string, error) {
+	is, err := getGlobalInfoSyncer()
+	if err != nil {
+		return "", err
+	}
+
+	// if the cache of prometheusAddr is over 5s, update the prometheusAddr
+	if time.Since(is.modifyTime) < TablePrometheusCacheExpiry {
+		return is.prometheusAddr, nil
+	}
+	is.prometheusAddr = ""
+	return is.getPrometheusAddr()
+}
+
+type prometheus struct {
+	IP         string `json:"ip"`
+	BinaryPath string `json:"binary_path"`
+	Port       int    `json:"port"`
+}
+
+type metricStorage struct {
+	PdServer struct {
+		MetricStorage string `json:"metric-storage"`
+	} `json:"pd-server"`
+}
+
+func (is *InfoSyncer) getPrometheusAddr() (string, error) {
+	// Get PD servers info.
+	pdAddrs := is.etcdCli.Endpoints()
+	if len(pdAddrs) < 0 {
+		return "", errors.New("pd unavailable")
+	}
+	var res string
+
+	// get prometheus address from pdApi
+	var url string
+	if strings.HasPrefix(pdAddrs[0], "http://") {
```

Comment:
Why need this check?

Reply:
When run the server, after a little while, the log would print a 
` [WARN] [config_handler.go:169] ["RPC to PD error when registering"] [error="rpc error: code = Unimplemented desc = unknown service configpb.Config"]`,
it also happend in the master branch, then the pdAddrs[0] would begin with the `http://` prefix.